### ISSUE TITLE
RFC: Operationalizing BIDS Derivatives common principles

### ISSUE THREAD

**Issue by effigies** [bids-standard/bids-specification#1242]
While attempting to add validation rules for BIDS Derivatives, we've found there is room for differences of interpretation in https://bids-specification.readthedocs.io/en/latest/05-derivatives/01-introduction.html (source included below)

https://github.com/bids-standard/bids-specification/blob/73d9968b55ce9b87371522d7efa8cda57978e705/src/05-derivatives/01-introduction.md#L14-L71

To operationalize these, I see the following conclusions:

1) If the filename is a valid raw filename, then that file is raw and all raw BIDS rules apply to that file.
2) If the file has a valid raw suffix (but the filename is a valid derivative filename, e.g., with the `desc-` entity), then the REQUIRED metadata for that suffix becomes RECOMMENDED, and the RECOMMENDED metadata becomes OPTIONAL.

Whether or not this is the consensus position, I think it is worth writing the rules explicitly.

**Comment by tsalo** [comment]
I agree with your conclusions. I think these rules are going to be difficult to enforce, especially when we want to consider derivative datasets independently of the associated "raw" datasets, but it seems like that's just the way things are (until 2.0, potentially).

One other conclusion I took from that section is that metadata fields in derivative datasets can't restricted like they are for (some types of) raw datasets, except in cases where specific fields are explicitly forbidden. I can't remember which data types restricted metadata fields (MRI?), but validation will be affected for at leave some.

**Comment by christinerogers** [comment]
Is this resolved by the updates in discussion for BIDS Derivatives? @effigies

**Comment by effigies** [comment]
@christinerogers Are you referring to something specific? I don't believe this was discussed in the last year.

**Comment by christinerogers** [comment]
@effigies please update this post-Copenhagen, and ahead of the Bordeaux meeting.

**Comment by effigies** [comment]
Revisiting this issue pre-OHBM-2026, this is not the approach that the validator has taken so far. Currently the validator does not demote most metadata by default, so this remains an open issue and spec clarity would be appreciated.

I still think these are the correct conclusions and we should write them into the spec, but in order to actually make them work, we will either need schema-based mechanisms or hard-code a transformation in the validator.

What we are currently considering to hard-code in the validator is:

* If the dataset type (not file) is derivative, then check whether `dataset.DatasetType` appears in the selectors.
  * If it does, then assume that the rule applies exactly as written.
  * If not, assume it is a rule for raw data and reduce the requirement levels (REQUIRED -> RECOMMENDED; RECOMMENDED/OPTIONAL -> OPTIONAL).

The validator currently stores the matching file rule in the validation context for its own use, but this is not part of the schema context. It could be added and used to determine whether a file is raw or derived, and hence whether raw or derived validation applies to the file. Alternately, a `file_type: 'raw' | 'derived'` field could be added, leaving the implementation opaque to the schema.

I do not like using string matching on selectors (whether for `dataset.DatasetType` or `file_type`) to determine whether or not to modify requirement levels. I would rather be explicit in the schema about whether a metadata field is universally required, required for raw/recommended for derivative, etc. Example requirement levels could be `required`/`required_raw`/`recommended`/`recommended_raw`/`optional`.

If we are reconsidering requirement levels, this could also be an opportunity to sort recommended fields into warn/hint/etc. I recall @bendichter had some language he used relating to NWB, but I can't find where this discussion was recorded.
